src/main/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStep.java

```diff
@@ -107,6 +122,20 @@
             return null;
         }
 
+        /**
+         * Returns true if we're in a multibranch job - behavior may be slightly different when that's not the case.
+         *
+         * @param job The job this build belongs to.
+         * @return True if this is a multibranch job, false otherwise.
+         */
+        private boolean isMultibranch(Job<?,?> job) {
+            ItemGroup<?> parent = job.getParent();
+            if (!(parent instanceof WorkflowMultiBranchProject)) {
+                return false;
+            }
+            return true;
```

Comment:
Eminently reasonable. Will give you a few more minutes for further review comments before pushing. =)